rewrite-maven/src/main/java/org/openrewrite/maven/RemoveRedundantDependencyVersions.java

```diff
@@ -199,19 +207,32 @@ public Xml.Document visitDocument(Xml.Document document, ExecutionContext ctx) {
                                 return null;
                             }
                             // some other element is also declared (executions, configuration, dependencies…), so just remove the version
-                            return tag.withContent(ListUtils.map(tag.getContent(), c -> c == version ? null : c));
+                            return tag.withContent(withoutVersion(tag, version));
                         }
                     } else {
                         Plugin p = findPlugin(tag);
                         if (p != null && matchesGroup(p) && matchesArtifact(p) && matchesVersion(p)) {
                             Xml.Tag version = tag.getChild("version").orElse(null);
-                            return tag.withContent(ListUtils.map(tag.getContent(), c -> c == version ? null : c));
+                            return tag.withContent(withoutVersion(tag, version));
                         }
                     }
                 }
                 return super.visitTag(tag, ctx);
             }
 
+            private @Nullable List<? extends Content> withoutVersion(Xml.Tag tag, Xml.@Nullable Tag version) {
+                return ListUtils.map(tag.getContent(), c -> {
+                    if (c == version) {
+                        if (((Xml.Tag) c).getValue().isPresent() && ((Xml.Tag) c).getValue().get().contains("${")) {
+                            final String propertyName = ((Xml.Tag) c).getValue().get().replace("${", "").replace("}", "");
```

Comment:
As indicated by Sam here https://github.com/openrewrite/rewrite/pull/5194#discussion_r2149906102 there could be multiple properties used

Reply:
Fix: https://github.com/openrewrite/rewrite/pull/5645/commits/fb7a0da5c650624585edafd602fc1ad482360d0a